doc/developer/platform/ux.md

```diff
@@ -1,20 +1,945 @@
+
+`SHOW INDEXES` and `mz_indexes` will gain an additional column specifying to
+which cluster the index belongs.
+
+There are no access control restrictions on individual indexes. Access to
+indexes is controlled by the permissions on the cluster containing the index.
+
+### Secret
+
+A secret is a new type of database object that securely stores a confidential
+credential.
+
+To create a new secret:
+
+```
+CREATE SECRET <name> AS '<secret>';
+```
+
+To drop an existing secret:
+
+```
+DROP SECRET <name>;
+```
+
+The existence of the secret will be stored in the catalog, but the actual
+contents of the secret will be securely stored in a secrets management system.
+
+Sources and sinks will need to learn to accept secret references in places that
+they currently expect an inline string or file path. Designing that new syntax
+is TODO.
+
+Referencing a secret in a source or sink definition requires that the user who
+creates the source or sink has `USAGE` privilege on the secret.
+
+There is intentionally no command to view a secret, though a sufficiently
+motivated attacker with the `USAGE` privilege could likely extract a secret with
+a well-constructed `CREATE SOURCE` command that references the secret.
+
+### Timeline
+
+TODO: this section is still relatively poorly understood. Are timelines 1:1
+with databases, or their own separate entity?
+
+TODO: engage @mjibson.
+
+TODO: add timelines to the overview diagram.
+
+UX thoughts from @frankmcsherry on timelines:
+
+> Within each REGION there may be multiple TIMELINEs, which correspond roughly
+> to coordinator timelines. A TIMELINE contains collections, and is the unit
+> within which it is possible to query across multiple collections.
+>
+> The main restrictions we plan to impose on users is that their access to data
+> is always associated with a REGION, CLUSTER, and TIMELINE, and attempts to
+> access data across these may result in poorer performance, or explicit query
+> rejection.
+
+### Role
+
+A role assigns a set of privileges to a group of users. Roles are scoped to a
+single region.
+
+Privileges can be granted on databases, schemas, and most database objects.
+There are also some special region-scoped privileges that control who can create
+certain types of objects, like `CREATE CLUSTER` and `CREATE SOURCE`. Details
+about what objects support what privileges are described in the sections above.
+
+The standard SQL `CREATE ROLE` statement will be used to map users to groups,
+and the standard SQL `GRANT` and `REVOKE` statements will be used to assign and
+unassign permissions to users and groups.
+
+## Assorted considerations
+
+### Version upgrades
+
+Services running outside the context of a region (e.g., the web UI and the
+global API) will be continuously deployed.
+
+Services running inside the context of a region will be upgraded to the latest
+version of Materialize Platform every week. Users will not be permitted to
```

Comment:
> every week

I'm absolutely in favor of what the paragraph says, but I wouldn't specify a weekly update cadence as a promise to our clients. It seems to me as an implementation detail that doesn't/shouldn't matter to users. Also, we might eventually determine a biweekly cadence suits us better...

Reply:
Ack, yeah, wasn't meant to be a promise! I've updated the text accordingly.